File: xadapter-sample/app/src/main/java/com/xadapter/sample/MainView.java
package com.xadapter.sample;

import java.util.List;

/**
 * by y.
 * <p>
 * Description:
 */
public interface MainView {

    void onChangeRootLayoutStatus(String status);

    void onRemoveAll();

    void onNetComplete(int type);

    void onNetError(int type);

    void onLoadNoMore();

    void onNetSuccess(List<Entity> entity);

    void onPagePlus();

    int getPage();

}

File: XAdapterSample/app/src/main/java/com/xadapter/sample/MainView.kt
package com.xadapter.sample

/**
 * by y.
 *
 *
 * Description:
 */
interface MainView {

    val page: Int

    fun onChangeRootLayoutStatus(status: String)

    fun onRemoveAll()

    fun onNetComplete(type: Int)

    fun onNetError(type: Int)

    fun onLoadNoMore()

    fun onNetSuccess(entity: List<Entity>)

    fun onPagePlus()

}
